hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java

```diff
@@ -827,10 +827,13 @@ static JpaOrder sortSpecification(SqmSelectStatement<?> sqm, Order<?> order) {
 	}
 
 	public static boolean isSelectionAssignableToResultType(SqmSelection<?> selection, Class<?> expectedResultType) {
-		if ( expectedResultType == null
-				|| selection != null && selection.getSelectableNode() instanceof SqmParameter ) {
+		if ( expectedResultType == null ) {
 			return true;
 		}
+		else if ( selection != null && selection.getSelectableNode() instanceof SqmParameter ) {
+			// assume the parameter type is not assignable to the result type
+			return false;
```

Comment:
Shouldn't it rather be this?

```suggestion
			return expectedResultType.isAssignableFrom( parameter.getAnticipatedType().getBindableJavaType() );
```

Reply:
Sure :+1: